### PR TITLE
Radio buttons for theme selection

### PR DESCRIPTION
**Pull request by MathisHage**
This PR improves the theme selection by the user.

### Changes :
* Added a `currentTheme` state flow in `AppThemeViewModel.kt`. It holds a new enum type `AppThemeValue` which can be either `LIGHT`, `DARK`, or `SYSTEM_DEFAULT`. The logic has been updated for the old `isDark` attribute to accordingly be `true` if the app should be in dark mode (`false` if light mode).
* Introduced a new alert dialog box when clicking on the "Theme" button in the settings screen. There are three radio buttons in this dialog : "Light", "Dark" and "System default". The changes are applied only when the "confirm" button is pressed (dismissing the dialog does not save the changes).
* Added new tests and updated old ones to stay up to date with these changes.

Here is a visual peek at the changes (my device is in dark mode) :

<p align="center"></p>

Waiting for your feedback ! Also, don't hesitate if there is missing documentation (especially for public methods) or if you find any bugs !

### CODE REVIEW

File: app/src/main/java/com/github/se/orator/model/theme/AppThemeViewModel.kt [hunk 1]
```diff
@@ -4,14 +4,28 @@ import android.content.Context
...
+}
+
 /**
  * ViewModel managing the current theme of the app.
  *
  * @param context The context of the app.
  */
 class AppThemeViewModel(private val context: Context) {
 
-  private val IS_DARK_NAME = "isDark"
```
Comment (helkhour) [inline app/src/main/java/com/github/se/orator/model/theme/AppThemeViewModel.kt:14]: Interesting approach introducing currentTheme as a StateFlow in AppThemeViewModel.

**Review by helkhour — APPROVED** [review]
I love how  this PR introduces a significant improvement to the theme selection feature with this new dynamic state management , the new dialog UI, and updated tests.  The use of AppThemeValue simplifies and makes the theme management more readable.
The separation of the currentTheme flow ensures clean state handling and aligns well with MVVM principles.
I like the use of RadioButton for theme selection, it makes the UI intuitive.

I would only suggest to add  more inline documentation to AppThemeViewModel to clarify the logic for currentTheme and its relationship with isDark.
Maybe also adding a test for edge cases where currentTheme might be null or invalid could be a good idea. 

Overall Great work ! I will approve the PR with minor suggestions.

**Comment by MathisHage** [comment]
Hey @helkhour, thanks for the review ! Next time, please don't approve if you have any suggestions ! Your remarks made me realize that there was no error handling mechanism when the memory value for `currentTheme` was invalid. Thanks for pointing that out !
As you suggested, I also added a test for this specific case, and clarified the documentation in `AppThemeViewModel.kt`.
